dali/pipeline/operators/paste/bbox_paste.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_CROP_BBOX_PASTE_H_
+#define DALI_PIPELINE_OPERATORS_CROP_BBOX_PASTE_H_
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+template <typename Backend>
+class BBoxPaste : public Operator<Backend> {
```

Comment:
You could  specialize it to CPUBackend directly here, since the run implem supports only a CPUBackend.

Reply:
Given the recent trend of adding GPU versions of bounding box processing stages, it seemed reasonable to keep it generic for future GPU variant.

Reply:
I thought that we had another implem for the GPU version
So you can template specialize RunImpl for CPUBackend here

Reply:
Specialization of the entire class is ok. What I don't want here is something like
`class BbFlip : public Operator<CPUBackend>`
Instead, we'd have

`template<>
class BBoxPaste<CPUBackend> : public Operator<CPUBackend>`

Is that what you had in mind?

Reply:
I was thinking about something similar as what we do here, specialize the implem in the .cc :
https://github.com/NVIDIA/DALI/blob/9ec7d0af11d1d207eee722a81c50f869a91f5a81/dali/pipeline/operators/crop/slice.cc#L59